CohortsScheduling/src/main/java/runnable/ScheduleRunnable.java

```diff
@@ -0,0 +1,108 @@
+package runnable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+
+import CohortDataClasses.Cohort;
+import CohortDataClasses.CohortSectionAssignment;
+import CohortDataClasses.Section;
+import CohortScoring.cohortScoring;
+import CohortsSolverData.CohortSolution;
+public class ScheduleRunnable implements Runnable {
+	private CohortSolution solutions[];
+	private boolean finished;
+	public void run() {
+		finished = false;
+		scheduleRunner(solutions);
+	}
+	
+	public boolean isFinished() {
+		return finished;
+	}
+	
+	public ScheduleRunnable(CohortSolution solutions[]) {
+		this.solutions = solutions;
+	}
+	
+	public static void scheduleRunner(CohortSolution solutions[])
+    {
+    	try {
+    		
+    		SolverFactory<CohortSolution> factory = SolverFactory.createFromXmlResource("SolverConfig.xml");
+    		Solver<CohortSolution> solver = factory.buildSolver();
+    		cohortScoring score = new cohortScoring();
+    		for(int i = 0; i<1 ;i++){
+    			Thread.currentThread();
+				if(Thread.interrupted())
+    				return;
+    			System.out.println("Before: " + score.calculateScore(solutions[i]));
+    			solutions[i] = (CohortSolution)solver.solve(solutions[i]);
+    			System.out.println("After: " + score.calculateScore(solutions[i]));
+    		}
+    		Thread.currentThread();
+			if(Thread.interrupted())
+				return;
+    		recordSolutions(solutions);
+    	}catch(Exception e) {
+    		e.printStackTrace();
+    	}
+    	
+    	
+    	
+    }
+
+	private static void recordSolutions(CohortSolution[] solutions) {
```

Comment:
this can be removed and replaced with a call to the DAO after each cohort finishes